packages/@aws-cdk/aws-ecs/lib/log-drivers/fluentd-log-driver.ts

```diff
@@ -0,0 +1,25 @@
+import { GenericLogDriver } from "./generic-log-driver"
+
+/**
+ * The configuration to use when creating a fluentd log driver.
+ */
+export interface FluentdLogDriverProps {
+  /**
+   * The configuration options to send to the log driver.
+   */
+  [key: string]: string;
```

Comment:
I think I would like more modeling here. If you don't know what the possible properties are, I'd rather avoid creating a class right now than just saying "random bag of options here".  Or is that an impossible task?

Reply:
> To my knowledge, the options section of the logdriver is purposely ambiguous

Yes, to the generic logdriver they'll be ambiguous. But every concrete logdriver (like awslogs) has a well-defined set of options, that we could guide and auto-complete the user towards.i

So for example, what are you likely to want to pass to a `journald` log driver?

Reply:
1. In case of defaults, we don't ENCODE them into the construct (so we simply don't output the option), but we generally favor annotating the property with a `@default` annotation, so that users have all the information they'll need right at their fingertips when their IDE autocompletes. Yes, there is a risk of it getting outdated, but we will have to trust in the power of the community and PRs to catch and correct these as they change.

2. Don't know, I'll have to see. How does the circulcar dependency error manifest itself?

3. The interface should feel natural to the user. If the argument is likely to be an AWS resource, I'd rather you accept the resource as an object and get, let's say, the ARN from it and write that to the options. If it's a number, I would prefer accepting the number and we stringify for the user.

4. In the vein of escape hatching, I'd like to keep the generic log driver around as well. It will be useful in one of two cases: (1) ECS adds a new log driver we don't have a class for yet and (2) in case we get one of the properties wrong, or they add a new property to a new log driver. The user experience won't be as nice, but at least they won't be blocked.

Reply:
Thanks for all the effort you've put into this by the way!

Reply:
Preferred way of stringification (to me) would be using TypeScript's string interpolation: `${this.props.someNumber}`.